File: src/main/scala/org/xbery/artbeams/articles/domain/EditedArticle.scala
package org.xbery.artbeams.articles.domain

import java.util.Date

import net.formio.binding.ArgumentName
import org.xbery.artbeams.common.assets.domain.EditedTimeValidity

/**
  * @author Radek Beran
  */
case class EditedArticle(
  @ArgumentName("id")
  id: String,
  @ArgumentName("externalId")
  externalId: Option[String],
  @ArgumentName("slug")
  slug: String,
  @ArgumentName("title")
  title: String,
  @ArgumentName("image")
  image: Option[String],
  @ArgumentName("perex")
  perex: String,
  @ArgumentName("bodyMarkdown")
  bodyMarkdown: String,
  @ArgumentName("validFrom")
  override val validFrom: Date,
  @ArgumentName("validTo")
  override val validTo: Option[Date],
  @ArgumentName("keywords")
  keywords: String,
  @ArgumentName("showOnBlog")
  showOnBlog: Boolean = false,
  @ArgumentName("showOnBlog")
  categories: java.util.List[String]
) extends EditedTimeValidity

File: src/main/kotlin/org/xbery/artbeams/articles/domain/EditedArticle.kt
package org.xbery.artbeams.articles.domain

import net.formio.upload.UploadedFile
import org.xbery.artbeams.common.assets.domain.EditedTimeValidity
import java.util.Date

/**
 * Editable article attributes.
 * @author Radek Beran
 */
data class EditedArticle(
    val id: String,
    val externalId: String?,
    val slug: String,
    val title: String,
    val image: String?,
    val file: UploadedFile?,
    val perex: String,
    val bodyEdited: String,
    val editor: String,
    override val validFrom: Date,
    override val validTo: Date?,
    val keywords: String,
    val showOnBlog: Boolean,
    val categories: List<String>,
) : EditedTimeValidity
